### PR TITLE
Smallfile cr params

### PR DESCRIPTION
**Pull request by bengland2**
do not merge yet.  add smallfile auto-pause parameter, plus passing all parameters from CR through to smallfile.

part of the fix for issue #595

### CODE REVIEW

**Comment by bengland2** [comment]
I 've tested with auto-pause and cleanup-delay-usec-per-file parameter using an OCS cluster..   Merging....

File: roles/smallfile/templates/smallfilejob.other_parameters.yaml.j2 [hunk 1]
```diff
@@ -1,6 +1,14 @@
 
 
 top: {{ smallfile_path }}/smallfile_test_data
+ 
+{% if workload_args.auto_pause is defined %}
+auto-pause: {{workload_args.auto_pause}}
```
Comment (sjug) [inline roles/smallfile/templates/smallfilejob.other_parameters.yaml.j2:6]: Missing spaces

File: roles/smallfile/templates/smallfilejob.other_parameters.yaml.j2 [hunk 2]
```diff
@@ -1,6 +1,14 @@
 
 
 top: {{ smallfile_path }}/smallfile_test_data
+ 
+{% if workload_args.auto_pause is defined %}
```
Comment (sjug) [inline roles/smallfile/templates/smallfilejob.other_parameters.yaml.j2:5]: Please add new args to `docs/smallfile.md`

File: roles/smallfile/templates/smallfilejob.other_parameters.yaml.j2 (hunk 1 above)
Comment (sjug) [reply roles/smallfile/templates/smallfilejob.other_parameters.yaml.j2:6]: Oddly there's a mix of spacing that exists in this file, any chance you could standardize the vars in this file to be all spaced correctly?

File: roles/smallfile/templates/smallfilejob.other_parameters.yaml.j2 [hunk 3]
```diff
@@ -15,16 +23,14 @@ files: {{ workload_args.files }}
 {% endif %}
 
 {% if workload_args.fsync is defined %}
-fsync: y
+fsync: {{ workload_args.fsync }}
```
Comment (sjug) [inline roles/smallfile/templates/smallfilejob.other_parameters.yaml.j2:26]: Given you're changing four different defaults, do we want to update the sample CRs `config/samples/smallfile/cr.yaml` (and/or in the docs) to add these parameters so we maintain  reasonable/functional defaults?

**Review by sjug — CHANGES_REQUESTED** [review]

**Comment by bengland2** [comment]
@sjug No problem, didn't notice the spacing issue, have been spoiled by pre-commit python linting ;-)    Also, you have a point about the example CRs, will do.   Thx for the review.

File: roles/smallfile/templates/smallfilejob.other_parameters.yaml.j2 (hunk 3 above)
Comment (bengland2) [reply roles/smallfile/templates/smallfilejob.other_parameters.yaml.j2:24]: so the full set of smallfile parameters is documented in smallfile github repo, don't have to repeat it.   This file defines a  subset of smallfile parameters that are usable with benchmark-operator.   The CI tests all the usable parameters now, I think.

File: roles/smallfile/templates/smallfilejob.other_parameters.yaml.j2 (hunk 1 above)
Comment (bengland2) [reply roles/smallfile/templates/smallfilejob.other_parameters.yaml.j2]: done and pushed

File: roles/smallfile/templates/smallfilejob.other_parameters.yaml.j2 (hunk 3 above)
Comment (sjug) [reply roles/smallfile/templates/smallfilejob.other_parameters.yaml.j2:24]: Right just doc the ones that we cover with the operator, I think the only two that are missing is the new `auto_pause` and `fsync`

File: roles/smallfile/templates/smallfilejob.other_parameters.yaml.j2 (hunk 2 above)
Comment (bengland2) [reply roles/smallfile/templates/smallfilejob.other_parameters.yaml.j2:3]: I resolved this differently, I linked to github smallfile repo.  Do not want to have to maintain in 2 or more places.  Also updated other parts of this doc.

**Comment by bengland2** [comment]
@sjug I answered your comments.  If I don't here otherwise I'll mark your comments as resolved.   Again thanks for your help, you pointed out some useful things.

File: roles/smallfile/templates/smallfilejob.other_parameters.yaml.j2 (hunk 3 above)
Comment (bengland2) [reply roles/smallfile/templates/smallfilejob.other_parameters.yaml.j2:24]: I also updated smallfile documentation to describe fsync and auto-pause and how to use them.

**Comment by bengland2** [comment]
BTW the reason that code coverage is bad above is that smallfile was removed from the CI for benchmark-operator, not my decision.

**Comment by bengland2** [comment]
I think it's ready to merge, comments have been addressed and replied to, marked as resolved.

**Review by rsevilla87 — APPROVED** [review]
LGTM

**Pull request merged**